### PR TITLE
PAYARA-1744 Boot commands now do envrionment variable subsitution

### LINKED ISSUE #1671: evaluate Environment Variables in Postbootcommandfile
# Description #
----------

I tried to use the new feature (payara full version) with 
`ENTRYPOINT asadmin start-domain --verbose --postbootcommandfile /opt/deployApp.cli`
In general it works, but there is no evaluation in the Postbootcommandfile.
This is now the feature request, so that this works:

parent Dockerfile (preparing my payara):

FROM alpine:latest
ADD payara/payara-4.1.2.172.zip lib/ojdbc7.jar target/classes/de/otto/verfuegbarkeit/payara/setupPayara.cli /opt/
RUN apk update && apk upgrade && \
    apk --no-cache add --update openjdk8 curl unzip tzdata && \
    ln -s /usr/lib/jvm/java-1.8-openjdk /opt/java && \
    rm -rf /var/cache/apk/* && \
    cp /usr/share/zoneinfo/Europe/Berlin /etc/localtime && echo "Europe/Berlin" > /etc/timezone && \
    unzip /opt/payara-4.1.2.172.zip -d /opt/ && rm /opt/payara-4.1.2.172.zip && mv /opt/payara41 /opt/payara && \
    rm -Rf /opt/payara/glassfish/domains/domain1 && \
    mv /opt/ojdbc7.jar /opt/payara/glassfish/domains/payaradomain/lib && \
    chmod a+rwx -R /opt/
ENV JAVA_HOME=/opt/java PATH="$PATH":/opt/java/jre/bin/ DEPLOYMENT_DIR=/opt/

WORKDIR /opt/payara/bin/
RUN /opt/payara/bin/asadmin start-domain && \
    /opt/payara/bin/asadmin multimode --file /opt/setupPayara.cli && \
	rm -f /opt/setup_payara.cli && \
    /opt/payara/bin/asadmin stop-domain
ENTRYPOINT asadmin start-domain --verbose --postbootcommandfile /opt/deployApp.cli
EXPOSE 8080 8181

Application Dockerfile:

FROM my-microservice
ENV application="myapp"
ADD target/${application}.war ${DEPLOYMENT_DIR}

in setupPayara.cli is sth like "create-jdbc-connection-pool ..." and file deployApp.cli looks like this:
`deploy /opt/${application}.war`
## Expected Outcome
the file deployApp.cli evalueates the environment variable from Application Dockerfile and the myapp.war is deployed.

## Current Outcome
File /opt/${application}.war cannot be found.

## Steps to reproduce (Only for bug reports) 
see above

## Environment ##

- **Payara Version**: 4.1.2.172
- **Edition**: full
- **JDK Version**: 8
- **Operating System**: alpine (Docker)
- **Database**: does'nt matter

### PR DESCRIPTION
**Pull request by Cousjava**
Fixes #1671

### CODE REVIEW

**Comment by Cousjava** [comment]
jenkins test please

**Comment by Cousjava** [comment]
Because that results in a circular dependency error

**Comment by Cousjava** [comment]
jenkins test please

File: nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/GlassFishMain.java [hunk 1]
```diff
@@ -276,6 +281,28 @@ private void doBootCommands(String file) {
             }
         }
         
+        
+        private String getEnvrironmentSubsitution(String value){
```
Comment (michaelranaldo) [inline nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/GlassFishMain.java]: Small note - typo'd method name

File: nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/GlassFishMain.java [hunk 2]
```diff
@@ -276,6 +281,28 @@ private void doBootCommands(String file) {
...
+                String newValue = System.getenv(matchValue);
+                if (newValue != null) {
+                    value = m2.replaceFirst(
+                            Matcher.quoteReplacement(m2.group(1) + newValue + m2.group(3)));
+                    m2.reset(value);
+                } 
+                i++;     
+            }
+            if (i >= MAX_SUBSTITUTION_DEPTH) {
+                Logger.getLogger(GlassFishMain.class.getName()).log(Level.SEVERE, "System property subsistution exceeded maximum of " + MAX_SUBSTITUTION_DEPTH);
```
Comment (michaelranaldo) [inline nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/GlassFishMain.java]: Minor typo in log output "substitution"

**Comment by Cousjava** [comment]
replaced by #1946

**Pull request closed without merging**